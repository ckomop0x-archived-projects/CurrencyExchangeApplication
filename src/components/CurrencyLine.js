import React, { Component } from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import TextField from 'material-ui/TextField';

class CurrencyLine extends Component {
	render() {
		const {
			floatingLabelText,
			value,
			onChange,
			onBlur
		} = this.props;
		return (
			<MuiThemeProvider muiTheme={getMuiTheme()}>
				<TextField
					style={{margin: 20}}
					floatingLabelText={floatingLabelText}
					type="number"
					value={value}
					onChange={onChange}
					onBlur={onBlur}
				/>
			</MuiThemeProvider>
		);
	}
}

export default CurrencyLine;