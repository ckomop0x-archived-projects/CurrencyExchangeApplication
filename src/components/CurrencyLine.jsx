import React from 'react';
import PropTypes from 'prop-types';
import { style } from 'typestyle';
import TextField from '@material-ui/core/TextField';

const CurrencyLine = ({ value, onChange }) => {
  const className = style({
    margin: 20,
    width: '90% !important',
    boxSizing: 'border-box'
  });

  return <TextField className={className} type="text" value={value} onChange={onChange} />;
};

CurrencyLine.propTypes = {
  floatingLabelText: PropTypes.string,
  value: PropTypes.node,
  onChange: PropTypes.func.isRequired
};

export default React.memo(CurrencyLine);
