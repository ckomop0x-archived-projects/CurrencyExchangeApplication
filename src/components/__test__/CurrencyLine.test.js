import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import CurrencyLine from '../CurrencyLine';

configure({ adapter: new Adapter() });

describe('<CurrencyLine />', () => {
  const onChange = jest.fn();
  const floatingLabelText = 'EUR';

  it('Should render without crashing', () => {
    const value = 0;
    const wrapper = shallow(
      <CurrencyLine onChange={onChange} floatingLabelText={floatingLabelText} value={value} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render with the value 100', () => {
    const value = 100;
    const wrapper = shallow(
      <CurrencyLine onChange={onChange} floatingLabelText={floatingLabelText} value={value} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
