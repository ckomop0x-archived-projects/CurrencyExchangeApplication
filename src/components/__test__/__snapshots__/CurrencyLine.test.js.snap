// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrencyLine /> Should render with the value 100 1`] = `ShallowWrapper {}`;

exports[`<CurrencyLine /> Should render without crashing 1`] = `ShallowWrapper {}`;
